### PR TITLE
deploy-huawei-app-gallery 1.0.0

### PR DESCRIPTION
**Pull request by pawelcala**
TagCheck

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

### CODE REVIEW

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  apt_get:
+  - name: git
+  - name: wget
+  - name: jq
+is_requires_admin_user: true
+is_always_run: false
+is_skippable: false
+run_if: ""
+inputs:
+- apk_path: null
```
Comment (koral--) [inline steps/deploy-huawei-app-gallery/1.0.0/step.yml]: It should probably default to `$BITRISE_APK_PATH`

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    is_required: true
+    title: App Id
+- huawei_client_id: null
+  opts:
+    is_expand: true
+    is_required: true
+    title: Client Id
+- huawei_client_secret: null
+  opts:
+    is_expand: true
```
Comment (koral--) [inline steps/deploy-huawei-app-gallery/1.0.0/step.yml]: It should be probably sensitive

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+- android
+type_tags:
+- deploy
+toolkit:
+  bash:
+    entry_file: step.sh
+deps:
+  brew:
+  - name: git
+  - name: wget
```
Comment (koral--) [inline steps/deploy-huawei-app-gallery/1.0.0/step.yml]: You are not using wget but curl in script

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+project_type_tags:
+- android
+type_tags:
+- deploy
+toolkit:
+  bash:
+    entry_file: step.sh
+deps:
+  brew:
+  - name: git
```
Comment (koral--) [inline steps/deploy-huawei-app-gallery/1.0.0/step.yml]: Git does not seem to be used in script

**Comment by koral--** [comment]
> * I have a test for my Step

Working test does not seem to be present in repo.

**Comment by pawelcala** [comment]
> * I have a test for my Step

@koral-- Sorry I misunderstood this step. I won't be able to attach test credentials to Huawei App Gallery. Any hint how can I add test for this?

**Comment by koral--** [comment]
You can store credentials in `.bitrise.secrets.yml` locally (and as secrets on bitrise.io if you have CI for this step). Don't forget to add this file to .gitignore.

More info: https://devcenter.bitrise.io/bitrise-cli/secrets/

Example of step with private credentials required for tests: https://github.com/DroidsOnRoids/bitrise-step-update-jira-issue-field/blob/develop/bitrise.yml

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+support_url: https://github.com/FutureMind/bitrise-step-huawei-app-gallery-apk-distribution/issues
+published_at: 2020-01-08T16:58:18.663184+01:00
+source:
+  git: https://github.com/FutureMind/bitrise-step-huawei-app-gallery-apk-distribution.git
+  commit: fe29d259c54362ac4db1fbb7cf8f2a4e770c2ee6
+host_os_tags:
+- osx-10.10
+- ubuntu-16.04
+project_type_tags:
+- android
```
[the remark this answers is not in the document]
Comment (pawelcala) [reply steps/deploy-huawei-app-gallery/1.0.0/step.yml]: Added new types but still not sure about this. I'm going to wait for users feedback as I can't test other types than Android.

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml (hunk 4 above)
Comment (pawelcala) [reply steps/deploy-huawei-app-gallery/1.0.0/step.yml]: Removed

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml (hunk 3 above)
Comment (pawelcala) [reply steps/deploy-huawei-app-gallery/1.0.0/step.yml]: Removed

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+- deploy
+toolkit:
+  bash:
+    entry_file: step.sh
+deps:
+  brew:
+  - name: git
+  - name: wget
+  - name: jq
+  apt_get:
```
[the remark this answers is not in the document]
Comment (pawelcala) [reply steps/deploy-huawei-app-gallery/1.0.0/step.yml]: Removed

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml (hunk 1 above)
Comment (pawelcala) [reply steps/deploy-huawei-app-gallery/1.0.0/step.yml]: Added BITRISE_APK_PATH as default. AAB is not supported by App Gallery.

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+is_requires_admin_user: true
+is_always_run: false
+is_skippable: false
+run_if: ""
+inputs:
+- apk_path: null
+  opts:
+    is_expand: true
+    is_required: true
+    title: APK path
```
[the remark this answers is not in the document]
Comment (pawelcala) [reply steps/deploy-huawei-app-gallery/1.0.0/step.yml]: Added descriptions

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml (hunk 2 above)
Comment (pawelcala) [reply steps/deploy-huawei-app-gallery/1.0.0/step.yml]: Added is_sensitive property

File: steps/deploy-huawei-app-gallery/1.0.0/step.yml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  opts:
+    is_expand: true
+    is_required: true
+    title: Client Id
+- huawei_client_secret: null
+  opts:
+    is_expand: true
+    is_required: true
+    title: Client Secret
+- lang: null
```
[the remark this answers is not in the document]
Comment (pawelcala) [reply steps/deploy-huawei-app-gallery/1.0.0/step.yml]: en-EN set as default

**Comment by pawelcala** [comment]
I've fixed all comments but I thinks I've also messed things up using rebase. Could you please recheck?

**Comment by lpusok** [comment]
@pawelcala It seems there are multiple step.yml-s in the PR, can you check please?

**Comment by godrei** [comment]
Hi @pawelcala ,
only 1 file change is allowed for a StepLib PR: https://github.com/bitrise-io/bitrise-steplib#important

Please reshare your latest step version:
- close this PR
- remove the `deploy-huawei-app-gallery-1.0.0` from your steplib fork
- run the [share-this-step](https://github.com/bitrise-io/bitrise-plugins-step/blob/master/create/templates/bitrise.yml.gotemplate#L62) workflow again

**Pull request closed without merging**